src/vs/workbench/services/untitled/common/untitledTextEditorModel.ts

```diff
@@ -93,7 +93,9 @@ export class UntitledTextEditorModel extends BaseTextEditorModel implements IUnt
 		// we have no associated file path. In that case we
 		// prefer the file name as title.
 		if (this.configuredLabelFormat === 'content' && !this.hasAssociatedFilePath && this.cachedModelFirstLineWords) {
-			return this.cachedModelFirstLineWords;
+			// Normalize multiple whitespaces to a single whitespace for the untitled editor label
+			const normalizedFirstLineWords = this.cachedModelFirstLineWords ? this.cachedModelFirstLineWords.replace(/\s+/g, ' ') : '';
```

Comment:
Why is this not done at the point where we cache the name?

https://github.com/microsoft/vscode/blob/50d9f05334aee7bbc416f675cceccd9a174de8c7/src/vs/workbench/services/untitled/common/untitledTextEditorModel.ts#L369-L372

Reply:
Update: Normalized when cached the first line words.